pkg/controller/scyllacluster/resource/resource_test.go

```diff
@@ -314,3 +321,590 @@ func TestMemberService(t *testing.T) {
+			"scylla/cluster":               "basic",
+			"scylla/datacenter":            "dc",
+			"scylla/rack":                  "rack",
+		}
+	}
+
+	newBasicStatefulSetLabelsWithVersion := func() map[string]string {
+		m := newBasicStatefulSetLabels()
+		m["scylla/scylla-version"] = ""
+		return m
+	}
+
+	newBasicStatefulSet := func() *appsv1.StatefulSet {
+		return &appsv1.StatefulSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        "basic-dc-rack",
+				Labels:      newBasicStatefulSetLabelsWithVersion(),
+				Annotations: nil,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion:         "scylla.scylladb.com/v1",
+						Kind:               "ScyllaCluster",
+						Name:               "basic",
+						UID:                "the-uid",
+						Controller:         pointer.BoolPtr(true),
+						BlockOwnerDeletion: pointer.BoolPtr(true),
+					},
+				},
+			},
+			Spec: appsv1.StatefulSetSpec{
+				Replicas: pointer.Int32Ptr(0),
+				Selector: &metav1.LabelSelector{
+					MatchLabels: newBasicStatefulSetLabels(),
+				},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: newBasicStatefulSetLabelsWithVersion(),
+						Annotations: map[string]string{
+							"prometheus.io/port":   "9180",
+							"prometheus.io/scrape": "true",
+						},
+					},
+					Spec: corev1.PodSpec{
+						Volumes: []corev1.Volume{
+							{
+								Name: "shared",
+								VolumeSource: corev1.VolumeSource{
+									EmptyDir: &corev1.EmptyDirVolumeSource{},
+								},
+							},
+							{
+								Name: "scylla-config-volume",
+								VolumeSource: corev1.VolumeSource{
+									ConfigMap: &corev1.ConfigMapVolumeSource{
+										LocalObjectReference: corev1.LocalObjectReference{
+											Name: "scylla-config",
+										},
+										Optional: pointer.BoolPtr(true),
+									},
+								},
+							},
+							{
+								Name: "scylla-agent-config-volume",
+								VolumeSource: corev1.VolumeSource{
+									Secret: &corev1.SecretVolumeSource{
+										SecretName: "scylla-agent-config-secret",
+										Optional:   pointer.BoolPtr(true),
+									},
+								},
+							},
+							{
+								Name: "scylla-client-config-volume",
+								VolumeSource: corev1.VolumeSource{
+									Secret: &corev1.SecretVolumeSource{
+										SecretName: "scylla-client-config-secret",
+										Optional:   pointer.BoolPtr(true),
+									},
+								},
+							},
+							{
+								Name: "scylla-agent-auth-token-volume",
+								VolumeSource: corev1.VolumeSource{
+									Secret: &corev1.SecretVolumeSource{
+										SecretName: "basic-auth-token",
+									},
+								},
+							},
+						},
+						InitContainers: []corev1.Container{
+							{
+								Name:            "sidecar-injection",
+								ImagePullPolicy: "IfNotPresent",
+								Command: []string{
+									"/bin/sh",
+									"-c",
+									fmt.Sprintf("cp -a /usr/bin/scylla-operator %s", naming.SharedDirName),
```

Comment:
don't use external variables like `naming.SharedDirName` in expected data, if the variable is change the change here should be explicit so the fallout is know and better catched on reviews